### PR TITLE
MM-23 server criar events, customer e business table e renomear auth 

### PR DESCRIPTION
**Pull request by mendes-develop**
## Summary by CodeRabbit

- **New Features**
	- Introduced new endpoints for user authentication: registration, login, and logout.
	- Added functionality for creating events in the application.
	- Implemented a new router for personal-related API endpoints, ensuring authentication is enforced.
	- Enhanced the dashboard and header components with updated functionalities and imports.

- **Bug Fixes**
	- Adjusted logout functionality to ensure proper handling of user sessions.

- **Documentation**
	- Updated OpenAPI documentation for new and existing routes.

- **Tests**
	- Added new test cases for authentication and environment variables to ensure reliability.

- **Chores**
	- Updated dependencies and file organization for better structure and maintainability.

### CODE REVIEW

File: packages/server/src/routes/auth/index.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { app } from "../../base/base-app";
+import { loginHandler, loginSpec } from "./login";
+import { logoutHandler, logoutSpec } from "./logout";
+import { registerHandler, registerSpec } from "./register";
+
+export const authRouter = app
+	.openapi(loginSpec, loginHandler)
+	.openapi(registerSpec, registerHandler)
+	.openapi(logoutSpec, logoutHandler);
```
[the remark this answers is not in the document]
Comment (mendes-develop) [reply packages/server/src/routes/auth/index.ts]: isso eh importante! Fazer um ticket pra rate limiter

File: packages/server/src/drizzle/schema.ts [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+import { z } from "zod";
+
+const Initial = z.string().min(2).max(2).toUpperCase();
+const genEntityId = (initials: string) =>
+	`${Initial.parse(initials)}_${createId().toUpperCase()}`;
 
 export const authTable = sqliteTable("auth", {
 	id: text("id", { length: 128 })
-		.$defaultFn(() => `AU_` + createId())
+		.$defaultFn(() => genEntityId("AU") + createId())
```
[the remark this answers is not in the document]
Comment (mendes-develop) [reply packages/server/src/drizzle/schema.ts]: verdade

File: packages/server/src/routes/personal/index.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { app } from "../../base/base-app";
+import { authMiddleware } from "../middleware/authentication";
+import { personalRoute, personalHandler } from "./me";
```
[the remark this answers is not in the document]
Comment (mendes-develop) [reply packages/server/src/routes/personal/index.ts]: this is a package, we cant use absolute path

File: packages/server/src/routes/auth/register.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+		passwordDigest,
+		name,
+		phone,
+	});
+	if (!newAuthUser) return c.json({ error: "User creation failed" }, 400);
+
+	const token = await generateToken(newAuthUser, JWT_SECRET_KEY);
+
+	setCookie(c, COOKIES.USER_ID, newAuthUser.email);
+	setCookie(c, COOKIES.USER_TOKEN, token);
```
[the remark this answers is not in the document]
Comment (mendes-develop) [reply packages/server/src/routes/auth/register.ts]: criar uma task pra fazer depois

File: packages/server/src/routes/auth/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mendes-develop) [reply packages/server/src/routes/auth/index.ts]: add these tasks for later

**Review by mendesmariosatech — APPROVED** [review]

**Pull request merged**